File: src/models/identity.rs
use serde::Serialize;

#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
pub struct IdentityProfileView {
    pub identity_id: String,
    pub display_name: Option<String>,
    pub avatar_hash: Option<String>,
    pub bio: Option<String>,
    pub stats_visibility: String,
    pub wallet_count: u32,
    pub created_at: i64,
    pub updated_at: i64,
    pub last_synced_block: i64,
    pub profile_version: i32,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
pub struct WalletLinkView {
    pub wallet_address: String,
    pub link_type: String,
    pub proof_signature: String,
    pub created_at: i64,
    pub verified_at: Option<i64>,
    pub last_synced_block: i64,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
pub struct IdentitySearchResult {
    pub identity_id: String,
    pub display_name: Option<String>,
    pub stats_visibility: String,
    pub updated_at: i64,
}

File: ts-models/src/identity.ts
import type { UnixSeconds } from './common.js';

export interface IdentityProfileView {
  readonly identity_id: string;
  readonly display_name: string | null;
  readonly avatar_hash: string | null;
  readonly bio: string | null;
  readonly stats_visibility: string;
  readonly wallet_count: number;
  readonly created_at: UnixSeconds;
  readonly updated_at: UnixSeconds;
  readonly last_synced_block: number;
  readonly profile_version: number;
}

export interface WalletLinkView {
  readonly wallet_address: string;
  readonly link_type: string;
  readonly proof_signature: string;
  readonly created_at: UnixSeconds;
  readonly verified_at: UnixSeconds | null;
  readonly last_synced_block: number;
}

export interface IdentitySearchResult {
  readonly identity_id: string;
  readonly display_name: string | null;
  readonly stats_visibility: string;
  readonly updated_at: UnixSeconds;
}

export interface IdentitySearchResponse {
  readonly query: string;
  readonly limit: number;
  readonly results: readonly IdentitySearchResult[];
}

export interface WalletVerificationRequest {
  readonly wallet_address: string;
  /** Optional proof signature (hex). */
  readonly signature: string | null;
}

export interface WalletVerificationResponse {
  readonly identity_id: string;
  readonly wallet_address: string;
  readonly linked: boolean;
  readonly verified: boolean;
  readonly proof_signature: string | null;
  readonly verified_at: UnixSeconds | null;
  readonly last_synced_block: number | null;
  readonly reason: string | null;
}
